### ISSUE TITLE
Question about input pointcloud

### ISSUE THREAD

**Issue by enricocovili** [url-kaist/patchwork-plusplus#62]
Hi, first let me congratulate to you for your fantastic work.

I want to ask a simple question, just to be 100% sure: does patchworkpp support only 360° pointclouds? 
I was trying to get it to work with a partial pointcloud but i don't think that's possible

**Comment by enricocovili** [comment]
Working with 180° pointcloud, my bad

**Comment by enricocovili** [comment]
I recently discovered an issue with a specific pointcloud. The crashes appears immediately when processing point cloud data from an OS1 128 Ouster LiDAR. Specifically, the crash occurs when the point cloud data does not begin from grade 0 of the LiDAR sensor.
I'm wondering if there's a configuration parameter I can adjust to resolve this, or would it be more practical to preprocess the point cloud data before passing it to Patchworkpp (like changing tf or the pointcloud itself) ?
thank you

**Comment by LimHyungTae** [comment]
Have you tried to run Patchwork++ on your data? I guess it does not count.

**Comment by enricocovili** [comment]
> Have you tried to run Patchwork++ on your data? I guess it does not count.

Yeah, my data is the pointcloud coming from the image above. We managed to find what seems to be the real issue:

https://github.com/url-kaist/patchwork-plusplus/blob/bef40cd5d3dc8c4d5853a5c85d59884e13195e15/cpp/patchworkpp/src/patchworkpp.cpp#L489
The code crash on this line while trying to access normal_, as it has not been initialized after patchwork wasn't able to find a ground. Our current """fix""" was to change all 4 vectors (https://github.com/url-kaist/patchwork-plusplus/blob/bef40cd5d3dc8c4d5853a5c85d59884e13195e15/cpp/patchworkpp/include/patchwork/patchworkpp.h#L179) to a fixed sized in the header. Everything seems to work well with this patch.

I'm reporting this as probably, almost surely, there is a better method to accomplish this task.

**Comment by LimHyungTae** [comment]
Aha, by any chance, did you change the sensor height parameter?

**Comment by enricocovili** [comment]
> Aha, by any chance, did you change the sensor height parameter?

Yeah, i'm not 100% sure but i think we changed it to 1m

**Comment by LimHyungTae** [comment]
But it's weird that there are none of the ground points. Could you share some pcd files for us?

**Comment by enricocovili** [comment]
> But it's weird that there are none of the ground points. Could you share some pcd files for us?

In the first screenshot there were no ground point because patchwork was crashing instantly. After the changes i described earlier everything was working fine (we are still tuning some parameters)

**Comment by enricocovili** [comment]
Any update on this? Can I help in some other ways?

**Comment by LimHyungTae** [comment]
That's strange, though. If you have time, could you do PR about that vector one?

So you'are saying you changed the original code:

```cpp
    Eigen::VectorXf normal_;
    Eigen::VectorXf singular_values_;
    Eigen::Matrix3f cov_;
    Eigen::VectorXf pc_mean_;
```
    
to 
    
```cpp
    Eigen::Vector3f normal_;
    Eigen::Vector3f singular_values_;
    Eigen::Matrix3f cov_;
    Eigen::Vector3f pc_mean_;
```

? Do I follow you?

**Comment by enricocovili** [comment]
Yeah you got the idea, i opened the PR as you asked

**Comment by enricocovili** [comment]
Any update?

**Comment by LimHyungTae** [comment]
Oh man I was so busy. Really sorry. I set a comment on that in #66

**Comment by LimHyungTae** [comment]
Solved by #81. Issue is closed.

The crash on partial (non-360°) point clouds — including the OS1 128 Ouster case you reported — was caused by `Eigen::VectorXf` (dynamic-size) members being accessed at fixed indices before being sized. Switching `normal_`, `singular_values_`, and `pc_mean_` to `Eigen::Vector3f` resolves it. Available on master and shipping in the next release.
